src/authorizations/components/redesigned/CustomApiTokenOverlay.tsx

```diff
@@ -81,23 +82,22 @@ const CustomApiTokenOverlay: FC<Props> = props => {
   }, [])
 
   useEffect(() => {
-    if (permissions['telegrafs'] && permissions['buckets']) {
-      return
-    }
-    const perms = {
-      otherResources: {read: false, write: false},
-    }
-    props.allResources.forEach(resource => {
-      if (resource === 'telegrafs') {
-        perms[resource] = props.telegrafPermissions
-      } else if (resource === 'buckets') {
-        perms[resource] = props.bucketPermissions
-      } else {
-        perms[resource] = {read: false, write: false}
+    if (!isEmpty(props.bucketPermissions.sublevelPermissions)) {
+      const perms = {
+        otherResources: {read: false, write: false},
       }
-    })
-    setPermissions(perms)
-  }, [props.telegrafPermissions, props.bucketPermissions])
+      props.allResources.forEach(resource => {
+        if (resource === 'telegrafs') {
+          perms[resource] = props.telegrafPermissions
+        } else if (resource === 'buckets') {
+          perms[resource] = props.bucketPermissions
+        } else {
+          perms[resource] = {read: false, write: false}
+        }
+      })
+      setPermissions(perms)
+    }
+  }, [props.remoteDataState])
```

Comment:
what if the permissions change?  (or that doesn't happen, or if the remote data state changes then the permissions change?)

Reply:
so the reason why i added props.remoteDataState inside the dependency array is because while buckets and telegrafs are being fetched, the remoteDataState will start at `NotStarted` then `Loading` and then once we get the data, it updates to `Done`. Each time remoteDataState changes, the useEffect hook will be called. BUT, none of the code inside the hook will run until remoteDataState is Done. Only then will props.bucketPermissions.sublevelPermissions will have value aka won't be empty. In which case, we can finally update the permissions state. I hope that answers your questions

Reply:
it does!  please add that as a comment to the code; then it'll be good to go :)